libfwupdplugin/fu-common.c

```diff
@@ -2236,26 +2236,49 @@ fu_common_cpuid (guint32 leaf,
  * Uses CPUID to discover the CPU vendor and check if it is Intel.
  *
  * Return value: %TRUE if the vendor was Intel.
+ * Deprecated: 1.5.5: Use fu_common_get_cpu_vendor() instead.
  *
  * Since: 1.5.0
  **/
 gboolean
 fu_common_is_cpu_intel (void)
+{
+	return fu_common_get_cpu_vendor () == FU_CPU_VENDOR_INTEL;
+}
+
+/**
+ * fu_common_get_cpu_vendor:
+ *
+ * Uses CPUID to discover the CPU vendor.
+ *
+ * Return value: %TRUE if the vendor was AMD.
+ *
+ * Since: 1.5.5
+ **/
+FuCpuVendor
+fu_common_get_cpu_vendor (void)
 {
 	guint ebx = 0;
 	guint ecx = 0;
 	guint edx = 0;
 
-	if (!fu_common_cpuid (0x0, NULL, &ebx, &ecx, &edx, NULL))
-		return FALSE;
+	if (fu_common_cpuid (0x0, NULL, &ebx, &ecx, &edx, NULL)) {
```

Comment:
in the case of no `#ifdef` does the compiler let you do this?  I would think you need to have something in the clause.  IE switch it to 
```
if (!fu...)
   return FU_CPU_VENDOR_UNKNOWN;
#ifdef ...
//do stuff
#endif 
return  FU_CPU_VENDOR_UNKNOWN;
```

Reply:
The compiler seems happy enough with the empy if clause, although I agree it's a bit wacky. I'll just move the entire function minus the last lines into the ifdef.